### PR TITLE
Sanitized dataset

### PR DESCRIPTION
**Pull request by gsalzer**
- simplified `datasets.yaml` (see issue #81)
- f-strings (string interpolation) instead of concatenation
- color constants now defined in utils

### CODE REVIEW

File: smartBugs.py [hunk 1]
```diff
@@ -12,58 +12,45 @@
...
     files_to_analyze = []
 
     if conf.datasets:
         for dataset in conf.datasets:
             base_name = dataset.split('/')[0]
-            if is_remote_dataset(cfg_dataset, base_name):
-                remote_info = get_remote_dataset(cfg_dataset, base_name)
-                base_path = remote_info['local_dir']
+            dataset_info = cfg_dataset[base_name]
+            if isinstance(dataset_info,dict) and 'url' in dataset_info and 'local_dir' in dataset_info:
```
Comment (jff) [inline smartBugs.py:25]: I would prefer to have this check inside a method named `is_remote_dataset` because it's more obvious when reading the code.

**Review by jff — COMMENTED** [review]
Thanks for this! Not only it removes code unnecessarily complicated, but you also introduced global (and readable) variables for the colours.

If you agree, I would just reintroduce the method `is_remote_dataset` (now much simpler!)

[reviewpad-sig]: <>
[View on **Reviewpad**](https://beta.reviewpad.com/review/github.com/smartbugs/smartbugs/pull/82?codereviewid=ede78862-2820-4019-8360-21202c3755c4)

File: smartBugs.py (hunk 1 above)
Comment (gsalzer) [reply smartBugs.py]: I think the options are to encapsulate the logic completely, like `is_remote_info/get_remote_info` or to let the tests and assignments access the  data structure directly, as I suggested. Mixing it like
```
if is_remote_info(dataset_info):
       ... dataset_info['url'] ... dataset_info['local_dir'] ...
```
has the disadvantage of obfuscating correctness. When checking the conditions for accessing `dataset_info` as a dict with components `url` and `local_dir` explicitly, before accessing it, correctness is obvious.

Encapsulation is fine, but I found the name `get_remote_dataset` misleading, better use `get_remote_info` or something similar. In fact, I checked twice to make sure that I hadn't overlooked the instructions where the dataset is actually loaded. So we could have something like
```
if is_remote_inf(dataset_info):
    (url,local_dir) = get_remote_info(dataset_info)
```
I modified the code accordingly.

A really clean solution would hide the structure of the configuration, and of the dataset file, completely. But this is maybe too much of an overhead for the moment.

**Comment by gsalzer** [comment]
Sorry, I mistakenly included the "fix_newlines" of Solidity files from the master branch. I could try to reset everything, but the this particular commit shouldn't do harm to the bytecodes branch either.

**Pull request merged**